src/System.IO.FileSystem/src/System/IO/PathHelpers.Windows.cs

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace System.IO
+{
+    internal static partial class PathHelpers
+    {
+        // Trim trailing white spaces, tabs etc but don't be aggressive in removing everything that has UnicodeCategory of trailing space.
+        // String.WhitespaceChars will trim more aggressively than what the underlying FS does (for ex, NTFS, FAT).    
+        internal static readonly char[] TrimEndChars = { (char)0x9, (char)0xA, (char)0xB, (char)0xC, (char)0xD, (char)0x20, (char)0x85, (char)0xA0 };
```

Comment:
I just moved this from PathHelpers.cs, but I'm assuming because of the last two chars in the list (horizontal ellipses and non-breaking space) that don't have an ASCII representation.  I'd be fine if we wanted to change this to use the C# character instead for the others where that works, though we can do that separately if desired, and if we do, we should look to do something similar in Path.Windows.cs: https://github.com/dotnet/corefx/blob/master/src/System.Runtime.Extensions/src/System/IO/Path.Windows.cs#L19-L48